ng-dev/pr/merge/target-label.ts

```diff
@@ -27,19 +52,19 @@ export class InvalidTargetLabelError {
 }
 
 /** Gets the target label from the specified pull request labels. */
-export async function getTargetLabelFromPullRequest(
+export async function getMatchingTargetLabelForPullRequest(
   config: Pick<MergeConfig, 'noTargetLabeling'>,
-  labels: string[],
+  labelsOnPullRequest: string[],
+  allTargetLabels: TargetLabel[],
```

Comment:
Why not just have the `getTargetLabelsForActiveReleaseTrains()` call occur within this function?  If we are worried about calling it multiple times we could just memoize the results in some way.

Reply:
This is done to make the function more self-contained and less concerned with other specifics of the merge tooling. Also it allows us to conveniently use this for testing, where we previously retrieved all target labels for active release trains, and then patched the same method with the result

Reply:
Okay, that works for me